pkg/proxy/ipvs/proxier.go

```diff
@@ -614,24 +614,15 @@ func (em proxyEndpointsMap) unmerge(other proxyEndpointsMap) {
 }
 
 // CanUseIPVSProxier returns true if we can use the ipvs Proxier.
-// This is determined by checking if all the required kernel modules are loaded. It may
+// This is determined by checking if all the required kernel modules can be loaded. It may
 // return an error if it fails to get the kernel modules information without error, in which
 // case it will also return false.
 func CanUseIPVSProxier() (bool, error) {
-	// Find out loaded kernel modules
-	out, err := utilexec.New().Command("cut", "-f1", "-d", " ", "/proc/modules").CombinedOutput()
-	if err != nil {
-		return false, err
-	}
-
-	mods := strings.Split(string(out), "\n")
-	wantModules := sets.NewString()
-	loadModules := sets.NewString()
-	wantModules.Insert(ipvsModules...)
-	loadModules.Insert(mods...)
-	modules := wantModules.Difference(loadModules).List()
-	if len(modules) != 0 {
-		return false, fmt.Errorf("Failed to load kernel modules: %v", modules)
+	for _, kmod := range ipvsModules {
+		err := utilexec.New().Command("modprobe", "--dry-run", kmod).Run()
```

Comment:
1. Currently, kube-proxy is running in container - of course, kube-proxy can run outside of container. I think the output of `modprobe` exec inside container is empty?

2. Did you installed `modprobe` command? Sorry, I didn't see it.

3. I admit better documentation and comments are needed. @Lion-Wei Can you do that?

Reply:
@m1093782566 
Thanks for you comment.
1. By default kernel modules are not included in containers, so neither auto-load mechanism nor `modprobe` doesn't work out of the box.  A workaround is to mount the kube-proxy container with ` -v /lib/modules:/host/lib/modules:ro` (of cause, privileged mode is needed). See [discussions at StackOverflow](https://stackoverflow.com/questions/33013539/docker-loading-kernel-modules).
2. `modprobe` should be builtin in most Linux distributions, but I haven't checked if the kube-proxy image contains it.

Reply:
I admit that I am not sure which approach is more appropriate at the moment. Looking forward to the opinions from your sides.

Reply:
Additional info added below.